app/models/user.rb

```diff
@@ -63,6 +67,27 @@ def is_application?
     false
   end
 
+  # State helpers.
+  #
+  # A User model begins life in the "temp" state, and can then be claimed by another user
+  # who originated from an OAuth login. Upon it being claimed it will be removed and it's
+  # data merged with the claimant.
+  #
+  # A User can also be created by a one of the consumer applications as a stand-in
+  # for a person who has not yet (or may never) created an account.  In this case
+  # the User model will be in the "unclaimed" state.  When the User does signup, they
```

Comment:
You will have to make sure this routine is called when people sign in through Google/FB/etc. By doing that, the emails are automatically approved, so the ContactInfosConfirm handler is not used.

Maybe we could just remove those special cases from the code and make the ContactInfosConfirm handler always be called... (and at that point, I guess it becomes a routine?)